doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,950 @@
+fig = Figure(figsize=(8,4))
+ax = fig.subplots()
+bins = st.slider(value=20, min_value=10, max_value=30, step=1, label="Bins")
+ax.hist(data, bins=bins)
+
+st.pyplot(fig)
+```
+
+The app looks like
+
+![Streamlit Basic Interactivity Example](https://user-images.githubusercontent.com/42288570/243358295-c26e90c2-0053-4441-9b98-2426d6d1a8e1.gif)
+
+The entire script is rerun *top to bottom* when you change the `bins` slider.
+
+#### Panel Basic Interactivity Example
+
+```python
+import panel as pn
+import numpy as np
+
+from matplotlib.figure import Figure
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+
+def plot(data, bins):
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+data = np.random.normal(1, 1, size=100)
+bins = pn.widgets.IntSlider(value=20, start=10, end=30, step=1, name="Bins")
+bplot = pn.bind(plot, data, bins)
+
+pn.Column(bins, bplot).servable()
+```
+
+Only the `plot` function is rerun when you change the `bins` slider. You specify that
+by *binding* the `plot` function to the `bins` widget using `pn.bind`.
+
+The app looks like
+
+![Panel Basic Interactivity Example](https://user-images.githubusercontent.com/42288570/243358311-04ed5189-1a79-4932-97ea-00898c27f283.gif)
+
+You might notice that the Panel app updates much quicker and more smoothly than the Streamlit app.
+This is due to fundamental differences in architectures. With Panel you will be able to
+develop very smooth and performant apps.
+
+### Multiple Updates Example
+
+This example will show you how to migrate code that produces a single result
+and updates the UI multiple times during the code execution.
+
+#### Streamlit Multiple Updates Example
+
+```python
+import random
+import time
+
+import streamlit as st
+
+
+def calculation_a():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+def calculation_b():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+st.write("# Calculation Runner")
+
+option = st.radio("Which calculation would you like to perform?", ("A", "B"))
+
+st.write("You chose: ", option)
+if option == "A":
+    if st.button("Press to run calculation"):
+        with st.spinner("Running... Please wait!"):
+            time_start = time.perf_counter()
+            result = calculation_a()
+            time_end = time.perf_counter()
+            st.write(f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete""")
+
+    else:
+        st.write("Calculation A did not run yet")
+
+elif option == "B":
+    if st.button("Press to run calculation"):
+        with st.spinner("Running... Please wait!"):
+            time_start = time.perf_counter()
+            result = calculation_b()
+            time_end = time.perf_counter()
+            st.write(f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete""")
+    else:
+        st.write("Calculation B did not run yet")
```

Comment:
Didn't test this, but I think you can write it like this?

Also, to highlight diff between A and B, I changed the nums a bit.

```suggestion
import random
import time
from textwrap import dedent

import streamlit as st


def calculation_a():
    time.sleep(1.5)
    return random.randint(0, 100)


def calculation_b():
    time.sleep(3.5)
    return random.randint(-100, 0)


st.write("# Calculation Runner")
option = st.radio("Which calculation would you like to perform?", ("A", "B"))
st.write("You chose: ", option)
if st.button("Press to run calculation"):
    with st.spinner("Running... Please wait!"):
        time_start = time.perf_counter()
        result = calculation_a() if option == "A" else calculation_b()
        time_end = time.perf_counter()
        message = dedent(
            f"""
            Done!
            Result: {result}
            The function took {time_end - time_start:1.1f} seconds to complete
            """
        )
else:
    st.write(f"Calculation {option} did not run yet")
```

Reply:
I've used your suggestion but without the `dedent` function.